src/auto-evo/EditorAutoEvoRun.cs

```diff
@@ -21,6 +21,11 @@ protected override void GatherInfo(Queue<IRunStep> steps)
         var map = Parameters.World.Map;
         var worldSettings = Parameters.World.WorldSettings;
 
+        foreach (var entry in map.Patches)
+        {
+            steps.Enqueue(new GenerateMiche(entry.Value, new SimulationCache(worldSettings), worldSettings));
```

Comment:
Does the parallelism help here a lot? If not it might be overall better to force the steps to not run in parallel and share a single simulation cache between all of them.

Reply:
Well I was referring to GenerateMiche specifically. That should be a super quick step that just preps the miche for the future steps. I think the additional time is added by the mutation code checking far more mutations. It's a time / result tradeoff to some extent. I think I have a few ideas to get it to prune out some more sub-optimal mutations though. That should speed stuff up a bit.

Reply:
In case if the step doesn't benefit from parallelism, it would be then better to share the cache instances to avoid doing a bunch of recalculations on a single thread that had already done the work.